modules/grizzly/src/main/java/org/glassfish/grizzly/CloseReason.java

```diff
@@ -40,26 +40,41 @@
 
 package org.glassfish.grizzly;
 
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 
 /**
  * An object, describing the reason why {@link Connection} was closed.
  * 
  * @author Alexey Stashok
  */
 public class CloseReason {
-    private static final IOException LOCALLY_CLOSED;
-    private static final IOException REMOTELY_CLOSED;
+    private static IOException LOCALLY_CLOSED;
+    private static IOException REMOTELY_CLOSED;
     
     public static final CloseReason LOCALLY_CLOSED_REASON;
     public static final CloseReason REMOTELY_CLOSED_REASON;
     
     static {
         LOCALLY_CLOSED = new IOException("Locally closed");
         LOCALLY_CLOSED.setStackTrace(new StackTraceElement[0]);
-        
+
         REMOTELY_CLOSED = new IOException("Remotely closed");
         REMOTELY_CLOSED.setStackTrace(new StackTraceElement[0]);
+        
+        try {
+          ByteArrayOutputStream  buffer = new ByteArrayOutputStream();
+          new ObjectOutputStream (buffer).writeObject(LOCALLY_CLOSED);
+          LOCALLY_CLOSED = (IOException) new ObjectInputStream (new ByteArrayInputStream (buffer.toByteArray ())).readObject();
+          
+          new ObjectOutputStream (buffer).writeObject(REMOTELY_CLOSED);
+          REMOTELY_CLOSED = (IOException) new ObjectInputStream (new ByteArrayInputStream (buffer.toByteArray ())).readObject();
+        } catch (Exception e) {
+          // This was done to avoid classloader references through the backtrace transient field.
```

Comment:
this should propagate the exception. There will be further errors is this happens